src/lib/common.c

```diff
@@ -180,11 +187,16 @@ int crc32file(const char *filename, uint32_t *crc32result)
 	if (rc_minor) {
 		rc_minor = -errno;
 		CT_ERROR(rc_minor, "fclose failed on '%s'", filename);
+		if (buf) free(buf);
```

Comment:
Maybe move the `if (buf) free(buf);` above the `int rc_minor;`?

Something like this?

```c
if (buf) free(buf);
buf = NULL;

int rc_minor = fclose(file);
```